src/components/common/LikeButton.jsx

```diff
@@ -0,0 +1,50 @@
+import { useState } from "react";
+import likeImg from "../../assets/like.png";
+
+export default function LikeButton({
+  like,
+  updateVotes,
+  patchTarget,
+  targetId,
+}) {
+  const [selected, setSelected] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const modifier = like ? 1 : -1;
+
+  return (
+    <>
+      <button
+        onClick={() => {
+          if (!selected) {
+            setIsError(false);
+            const amount = 1 * modifier;
+            setSelected(true);
+            updateVotes(amount);
+            patchTarget(targetId, amount).catch((err) => {
+              setIsError(true);
+              setSelected(false);
+              updateVotes(amount * -1);
+            });
```

Comment:
there is a LOT going on here that would be nice to see extracted into it's own handleUpvote function